source/simple_sub_pub_demo.c

```diff
@@ -195,6 +198,14 @@ extern MQTTAgentContext_t xGlobalMqttAgentContext;
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief The buffer to hold the topic filter. The topic is generated at runtime
+ * by adding the task name.
+ */
+static char topicBuf[ mqttexampleSTRING_BUFFER_LENGTH ]; /* Must persist until publish ack received. */
```

Comment:
This isn't ideal. Since every task calls snprintf on this topic buffer, it can't be shared between tasks. This demo won't if more than one of these tasks are created.